packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -340,13 +355,40 @@ Future<void> buildGradleProject({
       // Fall through. Pretend it's v1, and just go for it.
     case FlutterPluginVersion.v1:
       return _buildGradleProjectV1(project, gradle);
-    case FlutterPluginVersion.managed:
-      // Fall through. Managed plugin builds the same way as plugin v2.
+    case FlutterPluginVersion.v3:
+      return _buildGradleProjectV3(project, gradle, buildInfo, target, isBuildingBundle);
     case FlutterPluginVersion.v2:
       return _buildGradleProjectV2(project, gradle, buildInfo, target, isBuildingBundle);
   }
 }
 
+// TODO(mklim): Fix the obvious bugs.
```

Comment:
If "obvious bugs" refers to anything beyond the 2 items below, would you mind enumerating them?  What is obvious to one person may not be so obvious to another.